backends/gstreamer/device_monitor.rs

```diff
@@ -0,0 +1,61 @@
+use atomic_refcell::AtomicRefCell;
+use gst::Caps;
+use gst::DeviceExt;
+use gst::DeviceMonitor as GstDeviceMonitor;
+use gst::DeviceMonitorExt;
+use gst::DeviceMonitorExtManual;
+
+use servo_media::{MediaDeviceInfo, MediaDeviceKind};
+
+pub struct GStreamerDeviceMonitor {
+    devices: AtomicRefCell<Option<Vec<MediaDeviceInfo>>>,
```

Comment:
is the atomic refcell necessary?

can we instead return a GSTDeviceMonitor and have `navigator.mediadevices` contain one via a trait object?

Reply:
Sorry, I can't see how returning a `GstDeviceMonitor` makes the atomic refcell go away. The atomic refcell is here just to allow caching the result of the `GStreamerDeviceMonitor.enumerate_devices` call. If you think it is preferable to do the caching on the `navigator.mediaDevices` side, we can move the cache there, but I don't see the need to return the extra trait object.

Reply:
Yeah I want to do the caching on the mediadevices side.

The trait object lets us return a DeviceMonitor object that contains the cache (and will only be accessed from one thread), but it's also fine if we do the caching in servo.

Might need tweaks once we decide to support devicechange.